editor/scene_create_dialog.cpp

```diff
@@ -131,6 +131,22 @@ void SceneCreateDialog::update_dialog() {
 	root_name = root_name_edit->get_text().strip_edges();
 	if (root_name.is_empty()) {
 		root_name = scene_name.get_file().get_basename();
+
+		// Set the placeholder text.
+		if (root_name.is_empty()) {
+			root_name_edit->set_placeholder(TTR("Leave empty to derive from scene name"));
+		} else {
+			// Respect the desired root node casing from ProjectSettings and ensure it's a valid node name.
+			String adjusted_root_name = Node::adjust_name_casing(root_name);
+			root_name = adjusted_root_name.validate_node_name();
+
+			bool has_invalid_characters = root_name != adjusted_root_name;
+			if (has_invalid_characters) {
+				update_error(node_error_label, MSG_WARNING, TTR("Invalid root node name characters replaced with \"_\"."));
+			}
+
+			root_name_edit->set_placeholder(root_name);
+		}
 	}
 
 	if (root_name.is_empty() || root_name.validate_node_name().size() != root_name.size()) {
```

Comment:
```suggestion
	if (root_name.is_empty() || root_name.validate_node_name() != root_name) {
```

Prior to #75760 `validate_node_name()` removed invalid characters rather than replacing them with underscores. So now this check is not working correctly.

![](https://github.com/godotengine/godot/assets/47700418/0099e427-0918-44e4-bc14-6324a702b655)

Reply:
Good catch! I think I'll add a `has_invalid_node_name_characters()` method to the `String` class, since I don't like how current approach relies on implementation details of `validate_node_name()`.

Reply:
Perhaps we need `is_valid_node_name()` method, in addition to `validate_node_name()`. This would be consistent with `is_valid_identifier()` and `validate_identifier()` pair (the second is not exposed to scripting, see #67701). But I think that this is more of a material for a separate PR. As part of this PR, the `name.validate_node_name() != name` check looks OK to me.